apps/financial-aid/api/src/app/modules/user/user.resolver.ts

```diff
@@ -1,79 +1,60 @@
-import { Parent, Query, ResolveField, Resolver } from '@nestjs/graphql'
+import { Context, Parent, Query, ResolveField, Resolver } from '@nestjs/graphql'
 import { Inject, UseGuards } from '@nestjs/common'
 
 import type { Logger } from '@island.is/logging'
 import { LOGGER_PROVIDER } from '@island.is/logging'
-import {
-  ApplicationState,
-  CurrentApplication,
-  HomeCircumstances,
-} from '@island.is/financial-aid/shared/lib'
 import type { User } from '@island.is/financial-aid/shared/lib'
-import {
-  CurrentGraphQlUser,
-  JwtGraphQlAuthGuard,
-} from '@island.is/financial-aid/auth'
 
 import { UserModel } from './user.model'
-import { UserService } from './user.service'
 
 import { CurrentApplicationModel } from '../application'
-import { environment } from '../../../environments'
 import { StaffModel } from '../staff/models'
+import { CurrentUser } from '../decorators'
+import { BackendAPI } from '../../../services'
+import { IdsUserGuard } from '@island.is/auth-nest-tools'
 
-@UseGuards(JwtGraphQlAuthGuard)
+@UseGuards(IdsUserGuard)
 @Resolver(() => UserModel)
 export class UserResolver {
   constructor(
-    private readonly userService: UserService,
     @Inject(LOGGER_PROVIDER)
     private readonly logger: Logger,
   ) {}
 
-  private fakeUsers: { [key: string]: CurrentApplication | null } = {
-    '0000000000': null,
-    '0000000001': {
-      id: 'dbfc6ff1-58e0-4815-8fa9-1e06ddace1c3',
-      state: ApplicationState.INPROGRESS,
-      homeCircumstances: HomeCircumstances.OWNPLACE,
-      usePersonalTaxCredit: true,
-      created: '2021-09-11 18:11:17.103+00',
-    },
-    '0000000003': {
-      id: '5ebdb6ca-edcb-4391-bda7-f5999d2b6b08',
-      state: ApplicationState.DATANEEDED,
-      homeCircumstances: HomeCircumstances.OWNPLACE,
-      usePersonalTaxCredit: true,
-      created: '2021-09-12 18:11:17.103+00',
-    },
+  private async handleNotFoundException<T>(callback: () => Promise<T>) {
+    try {
+      return await callback()
+    } catch (e) {
+      if (e.extensions.response.status === 404) {
+        return undefined
+      }
+      throw e
+    }
   }
 
   @Query(() => UserModel, { nullable: true })
-  async currentUser(
-    @CurrentGraphQlUser() user: User,
-  ): Promise<UserModel | undefined> {
+  async currentUser(@CurrentUser() user: User): Promise<UserModel | undefined> {
     this.logger.debug('Getting current user')
-
     return user as UserModel
   }
 
   @ResolveField('currentApplication', () => CurrentApplicationModel)
   async currentApplication(
     @Parent() user: User,
-  ): Promise<CurrentApplicationModel | null> {
-    // Local development
-    this.logger.debug(
-      `Getting current application for nationalId: ${user.nationalId}`,
+    @Context('dataSources') { backendApi }: { backendApi: BackendAPI },
+  ): Promise<CurrentApplicationModel | undefined> {
+    this.logger.debug('Getting current application for nationalId')
+    return await this.handleNotFoundException(() =>
```

Comment:
Is this error not handled in the backend? application.controller? 🤓

Reply:
Nope it is thrown there, originally I wanted to return undefined in the controller because `currentApplication` is optional. But when I threw `user.service.ts` into the trash and started to use `backend.ts` like every other route we use I ran into a problem that the controller returned undefined and then `apollo-datasource-rest` tries to run `.json()` on that value and then eats the error and returns an empty string, which I don't want. 

So then I decided to throw an error in the controller, if we have no current application and catch it in the resolver and return undefined for it. 

Up for discussion if we want to take another route 😄 

The reason I refactored this is I didn't like that all routes were going through the same flow except for routes used in this resolver.